lib/livebook/intellisense/docs.ex

```diff
@@ -107,28 +107,30 @@ defmodule Livebook.Intellisense.Docs do
           :protocol | :implementation | :exception | :struct | :behaviour | nil
   def get_module_subtype(module) do
     cond do
-      module_has_function?(module, :__protocol__, 1) ->
+      not ensure_loaded?(module) ->
+        nil
+
+      function_exported?(module, :__protocol__, 1) ->
         :protocol
 
-      module_has_function?(module, :__impl__, 1) ->
+      function_exported?(module, :__impl__, 1) ->
         :implementation
 
-      module_has_function?(module, :__struct__, 0) ->
-        if module_has_function?(module, :exception, 1) do
+      function_exported?(module, :__struct__, 0) ->
+        if function_exported?(module, :exception, 1) do
           :exception
         else
           :struct
         end
 
-      module_has_function?(module, :behaviour_info, 1) ->
+      function_exported?(module, :behaviour_info, 1) ->
         :behaviour
 
       true ->
         nil
     end
   end
 
-  defp module_has_function?(module, func, arity) do
-    Code.ensure_loaded?(module) and function_exported?(module, func, arity)
-  end
+  defp ensure_loaded?(Elixir), do: false
```

Comment:
```suggestion
  # In case insensitive file systems, attempting to load
  # Elixir will log a warning in the terminal as it wrongly
  # loads elixir.beam, so we explicitly list it.
  defp ensure_loaded?(Elixir), do: false
```